src/main/java/teammates/logic/api/EmailGenerator.java

```diff
@@ -397,6 +399,82 @@ private EmailWrapper generateFeedbackSessionEmailBaseForInstructorReminders(
                 EmailType.FEEDBACK_UNPUBLISHED.getSubject());
     }
 
+    public EmailWrapper generateFeedbackSessionResendLinksEmail(String userEmail) {
+        Instant endTime = new Date().toInstant();
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.MONTH, -6);
+        Instant startTime = calendar.getTime().toInstant();
+        String subject = EmailType.FEEDBACK_LINKS_RESENT.getSubject();
+        StringBuffer linksFragmentValue = new StringBuffer(1000);
+        String emailBody;
+        String studentName = "";
+
+        List<FeedbackSessionAttributes> sessions = fsLogic.getAllOpenFeedbackSessions(startTime, endTime);
+        for (FeedbackSessionAttributes session : sessions) {
+            CourseAttributes course = coursesLogic.getCourse(session.getCourseId());
+            StudentAttributes student = studentsLogic.getStudentForEmail(course.getId(), userEmail);
+
+            if (student != null) {
+                studentName = student.getName();
+                String submitUrlHtml = "(Feedback session is not yet opened)";
+                String reportUrlHtml = "(Feedback session is not yet published)";
+
+                if (session.isOpened()) {
+                    String submitUrl = Config.getAppUrl(Const.ActionURIs.STUDENT_FEEDBACK_SUBMISSION_EDIT_PAGE)
+                            .withCourseId(course.getId())
+                            .withSessionName(session.getFeedbackSessionName())
+                            .withRegistrationKey(StringHelper.encrypt(student.key))
+                            .withStudentEmail(student.email)
+                            .toAbsoluteString();
+                    submitUrlHtml = "<a href=\"" + submitUrl + "\">" + submitUrl + "</a>";
+                }
+
+                if (session.isClosed()) {
+                    submitUrlHtml = "(Feedback session is closed)";
+                }
```

Comment:
I think session.isClosed() is not exactly opposite to session.isOpen() as there is a graced period, so it would be better not to put them together right? Pls correct me if I am wrong.

Reply:
I see. I'm not entirely sure here either so let's leave it for final review later.